libs/api/src/main/java/de/cyface/api/Authorizer.java

```diff
@@ -0,0 +1,206 @@
+
+        final var roles = principal.getJsonArray(DEFAULT_ROLE_FIELD).stream().map(r -> new Role((String)r)).collect(Collectors.toList());
+        final var managerRoles = roles.stream().filter(r -> r.getType().equals(Role.Type.GROUP_MANAGER))
+                .collect(Collectors.toList());
+        final var username = Validate.notEmpty(principal.getString(DEFAULT_USERNAME_FIELD));
+
+        // Guest users and group users can only access their own account
+        if (managerRoles.isEmpty()) {
+            return Future.succeededFuture(Collections.singletonList(username));
+        } else if (managerRoles.size() > 1) {
+            return Future.failedFuture(new IllegalArgumentException(
+                    String.format("User %s is manager of more than one group.", username)));
+        } else {
+            final Promise<List<String>> promise = Promise.promise();
+            final var groupManager = managerRoles.get(0);
+            final var loadUsers = loadAccessibleUsers(groupManager);
+            loadUsers.onSuccess(groupUsers -> {
+                final var users = new ArrayList<>(groupUsers);
+                users.add(username);
+                promise.complete(users);
+            });
+            loadUsers.onFailure(promise::fail);
+            return promise.future();
+        }
+    }
+
+    /**
+     * Loads all users a specific group manager can access.
+     *
+     * @param groupManager The {@code Role} group manager to load the users for
+     * @return The usernames of the users which the {@code groupManager} can access
+     */
+    public Future<List<String>> loadAccessibleUsers(final Role groupManager) {
+        Validate.isTrue(groupManager.getType().equals(Role.Type.GROUP_MANAGER));
+        Validate.notEmpty(groupManager.getGroup());
+
+        final Promise<List<String>> promise = Promise.promise();
+        final var future = promise.future();
+        final var groupUsersRole = groupManager.getGroup() + DatabaseConstants.USER_GROUP_ROLE_SUFFIX;
+        mongoUserDatabase.find(DatabaseConstants.COLLECTION_USER,
+                new JsonObject().put(DEFAULT_ROLE_FIELD, groupUsersRole),
+                mongoResult -> {
+                    if (mongoResult.succeeded()) {
+                        final var users = mongoResult.result();
+                        final var userNames = users.parallelStream()
+                                .map(groupUser -> groupUser.getString(DEFAULT_USERNAME_FIELD))
+                                .collect(Collectors.toList());
+                        promise.complete(userNames);
+                    } else {
+                        promise.fail(mongoResult.cause());
+                    }
+                });
+        return future;
+    }
+
+    @SuppressWarnings("unused") // API
+    public MongoAuthentication getAuthProvider() {
+        return authProvider;
+    }
+
+    public MongoClient getMongoUserDatabase() {
+        return mongoUserDatabase;
+    }
```

Comment:
What is the reason for all the methods to be public? Are they a required part of the interface?